<?xml version="1.0"?>
<launch>
    <arg name="steps_per_phase" default="10"/>
    <arg name="plotting" default="true" />
    <arg name="file_write_cb" default="false" />
    <arg name="box_side" default="0.2"/>
    <arg name="box_x" default="0.15"/>
    <arg name="box_y" default="0.4"/>
    <arg name="box_parent_link" default="workcell_base"/>
    <arg name="rviz" default="true"/>
    <arg name="testing" default="false"/>

    <!-- Load Robot Model    -->
    <param name="robot_description"  command="$(find xacro)/xacro --inorder $(find tesseract_ros_examples)/urdf/pick_and_place_plan.xacro"/>

    <!-- Load Robot Semantic -->
    <param name="robot_description_semantic" textfile="$(find tesseract_ros_examples)/config/pick_and_place_plan.srdf" />

    <!-- Set simulated box parameters -->
    <param name="box_side" value="$(arg box_side)"/>
    <param name="box_x" value="$(arg box_x)"/>
    <param name="box_y" value="$(arg box_y)"/>
    <param name="box_parent_link" value="$(arg box_parent_link)"/>
    <group unless="$(arg testing)">

      <!-- Launch GUI to drive joints   -->
      <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />

      <!-- Launch robot state publisher - may need to remap to /iiwa/joints..-->
      <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

      <!-- Launch planner node -->
      <node name="tesseract_ros_examples_pick_and_place_example_node" pkg="tesseract_ros_examples" type="tesseract_ros_examples_pick_and_place_example_node" output="screen">
          <param name="steps_per_phase" value="$(arg steps_per_phase)"/>
          <param name="plotting" value="$(arg plotting)"/>
          <param name="file_write_cb" value="$(arg file_write_cb)" />
          <param name="rviz" type="bool" value="$(arg rviz)"/>
      </node>

      <!-- Launch visualization -->
      <node if="$(arg rviz)" pkg="rviz" type="rviz" name="tesseract_ros_examples_pick_and_place_example_rviz"
          args="-d $(find tesseract_ros_examples)/config/examples.rviz" />
    </group>

</launch>
